tap_slack/tap.py

```diff
@@ -68,11 +69,28 @@ class TapSlack(Tap):
             th.ArrayType(th.StringType),
             description="A list of channel IDs that should not be retrieved. Excluding overrides a selected setting, so if a channel is included in both selected and excluded, it will be excluded.",
         ),
+        th.Property(
+            "include_admin_streams",
+            th.BooleanType,
+            default=False,
+            description="Whether to include streams that require admin privileges or not. If the user does not have the proper scopes then the tap will throw and exception.",
+        ),
     ).to_dict()
 
     def discover_streams(self) -> List[Stream]:
         """Return a list of discovered streams."""
-        return [stream_class(tap=self) for stream_class in STREAM_TYPES]
+
+        if self.config.get("include_admin_streams"):
+            streams = [stream_class(tap=self) for stream_class in STREAM_TYPES]
+        else:
+            # Exclude admin streams
+            streams = [
+                stream_class(tap=self)
+                for stream_class in STREAM_TYPES
+                if stream_class not in ADMIN_STREAM_TYPES
+            ]
```

Comment:
Instead of having to put an admin stream class in both `STREAM_TYPES` and `ADMIN_STREAM_TYPES`, wdyt about making them mutually exclusive. So `IntegrationLogsStream` would only live in `ADMIN_STREAM_TYPES`.

If that's OK, then maybe the method can be simplified:

```suggestion
        streams = [stream_class(tap=self) for stream_class in STREAM_TYPES]

        if self.config.get("include_admin_streams"):
            streams.extent(stream_class(tap=self) for stream_class in ADMIN_STREAM_TYPES]
```

Reply:
@edgarrmondragon I love it